### DISCUSSION TITLE
azure blobs

### DISCUSSION THREAD

**Discussion by marco-santulli** [Budibase/budibase#3594] [Ideas]
Hi,

is there any plan to support azure blobs?  I have seen that there is a way minio could work as a gateway so maybe it is simple to do?

M

**Comment by mjashanks** [comment]
There are no plans to do this, though I could see us doing do in the future - as we get more cloud-provider specific deployment options. It won't be happening soon.

**Reply by marco-santulli** [reply]
@mjashanks I have seen you have given some suggestions on this in the past, but I am facing an issue that is "typical". Maybe you have some ideas.
We need to manage attachments that we currently store in a blob storage (it would be the same with other storage platforms).  We currently save a copy of the attachments in the BB internal DB so we can serve them to the user.

I would be happy to store in the DB (we use a postgres) an ID/url for the file, but I have the problem of then serving those attachments to the browser in a secure way (without a public link).  The ideal would be if we could have a functionality to make BB "proxy" the request.  That's why I was suggesting using server side functions in another post, but with the function returning the actual media.  Do you think this could be a solution?

M

**Reply by mjashanks** [reply]
I see what you are trying to do and don't think that there is a workaround at the moment.

The way that I would see this working in the future would be to 
a) have `Azure Blob Storage` as a datasource.
b) Support `Attachment` or `File` data types, in external datasources.

If implemented as a datasource, then all requests would be proxied through the server.

I can totally see that this is a typical issue - I think what you have described is important. It's simply a matter of priorities for us right now - we have to pick which things to work on first.

Sorry that I can't help you more with it right now.

**Reply by marco-santulli** [reply]
I mike, no worries, I understand.
I think that the server side functions might be a solution for this, don't you think?

**Reply by aderici** [reply]
there is a minio - azure blob gateway docker but i think it will be retired soon. Grab it before its gone
